mibidata/tiff.py

```diff
@@ -289,29 +286,49 @@ def _page_metadata(page, description):
     date = datetime.datetime.strptime(
         page.tags['datetime'].value.decode(ENCODING),
         _DATETIME_FORMAT)
-    return {
-        'run': description.get('mibi.run'),
+
+    # check version for backwards compatibility
+    _convert_from_previous(description)
+
+    metadata = {}
+    for key, val in description.items():
+        if key.startswith('mibi.'):
+            metadata[key[5:]] = val
+        elif key not in RESERVED_MIBITIFF_ATTRIBUTES:
+            metadata[key] = val
+
+    metadata.update({
         'coordinates': (
             _cm_to_micron(page.tags['x_position'].value),
             _cm_to_micron(page.tags['y_position'].value)),
         'date': date,
-        'size': size,
-        'slide': description.get('mibi.slide'),
-        'point_name': description.get('mibi.description'),
-        'folder': description.get('mibi.folder'),
-        'dwell': description.get('mibi.dwell'),
-        'scans': description.get('mibi.scans'),
-        'aperture': description.get('mibi.aperture'),
-        'instrument': description.get('mibi.instrument'),
-        'tissue': description.get('mibi.tissue'),
-        'panel': description.get('mibi.panel'),
-        'mass_offset': description.get('mibi.mass_offset'),
-        'mass_gain': description.get('mibi.mass_gain'),
-        'time_resolution': description.get('mibi.time_resolution'),
-        'miscalibrated': description.get('mibi.miscalibrated'),
-        'check_reg': description.get('mibi.check_reg'),
-        'filename': description.get('mibi.filename'),
-    }
+        'size': size})
+
+    return metadata
+
+
+def _convert_from_previous(description):
+    """Convert old metadata format for backwards compatibility.
+
+    Most of these conversions would happen during MibiImage construction,
+    but we do them here in case reading the info only.
+    """
+    if not description.get('mibi.fov_name') and description.get(
+            'mibi.description'):
+        description['mibi.fov_name'] = description.pop('mibi.description')
+    # TODO: Clean up repetition between this and the same MibiImage method
+    if description.get('mibi.folder') and not description.get('mibi.fov_id'):
+        description['mibi.fov_id'] = description['mibi.folder'].split('/')[0]
+        warnings.warn(
```

Comment:
I think I've seen this with other libraries when printing warnings, so I don't know if it's necessarily a bug here:

![image](https://user-images.githubusercontent.com/32075205/69266848-d0802580-0b80-11ea-87f2-b8c16cc41c55.png)

Reply:
This is a warning to the user to inform that he's trying to read old data, but we're converting it to the current format.